File: smart_lock_library/src/main/java/smart_lock_library/callback/ForgetCredentialsResultCallback.java
package smart_lock_library.callback;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.Log;

import com.google.android.gms.common.api.ResultCallback;
import com.google.android.gms.common.api.Status;

import armymau.it.smart_lock_library.R;
import smart_lock_library.activity.SmartLockActivity;
import smart_lock_library.utils.SmartLockConstants;

public class ForgetCredentialsResultCallback implements ResultCallback<Status> {

    private final Context context;

    public ForgetCredentialsResultCallback(Context context) {
        this.context = context;
    }

    @Override
    public void onResult(@NonNull Status status) {

        ((SmartLockActivity) context).hideProgress();

        if (status.isSuccess()) {
            onCredentialsForgotten();
        } else {
            onCredentialsForgotten();
            Log.e(SmartLockConstants.TAG, context.getResources().getString(R.string.error_forget_failure));

            //Snackbar.make(((SmartLockActivity) context).findViewById(android.R.id.content), R.string.error_forget_failure, Snackbar.LENGTH_SHORT).show();
        }
    }

    private void onCredentialsForgotten() {
        //Snackbar.make(((AppCompatActivity) context).findViewById(android.R.id.content), R.string.credentials_forgotten, Snackbar.LENGTH_SHORT).show();
        ((SmartLockActivity) context).onCredentialsForgottenListener();
    }
}

File: app/src/main/java/smart_lock_library_kt/callback/ForgetCredentialsResultCallback.kt
package smart_lock_library_kt.callback

import android.content.Context
import android.util.Log
import armymau.it.smart_lock_library.R
import com.google.android.gms.common.api.ResultCallback
import com.google.android.gms.common.api.Status
import smart_lock_library_kt.activity.SmartLockActivity
import smart_lock_library_kt.utils.TAG

class ForgetCredentialsResultCallback : ResultCallback<Status> {

    var context: Context

    constructor(context: Context) {
        this.context = context
    }

    override fun onResult(status: Status) {

        (context as SmartLockActivity).hideProgress()

        if (status.isSuccess) {
            onCredentialsForgotten()
        } else {
            onCredentialsForgotten()
            Log.e(TAG, context.resources.getString(R.string.error_forget_failure))

            //Snackbar.make(((SmartLockActivity) context).findViewById(android.R.id.content), R.string.error_forget_failure, Snackbar.LENGTH_SHORT).show();
        }
    }

    private fun onCredentialsForgotten() {
        //Snackbar.make(((AppCompatActivity) context).findViewById(android.R.id.content), R.string.credentials_forgotten, Snackbar.LENGTH_SHORT).show();
        (context as SmartLockActivity).onCredentialsForgottenListener()
    }
}